### PR TITLE
Add initial package structure

### PR DESCRIPTION
**Pull request by destogl**
Here initial package structure with:

* main packages
* README.md file
* enabled GitHub Actions
* other config files (e.g. .gitignore, etc.)

### CODE REVIEW

**Comment by olivier-stasse** [comment]
@destogl Should I close my PR ?

File: ros2_control_demos/package.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0"?>
+<package format="3">
+  <name>ros2_control_demos</name>
+  <version>0.0.0</version>
+  <description>Metapackage for ROS2 control demos packages</description>
+
+  <author email="ostasse@laas.fr">Olivier Stasse</author>
+  <author email="denis@stogl.de">Denis Štogl</author>
+
+  <maintainer email="allabana@amazon.com">Anas Abou Allaban</maintainer>
```
Comment (bmagyar) [inline ros2_control_demos/package.xml:10]: The guys from AWS won't take part in maintenance, please remove this line

**Comment by bmagyar** [comment]
@destogl you may need to add a dummy test that generates *some* report for codecov to pass

**Comment by destogl** [comment]
> Looking forward to some code, now that we've got 1k lines of boilerplate stuff already ;)

I will not any code here just, configurations for the repo :)

I think we have every thing now, just want to make all tests happy....

File: ros2_control_demos/package.xml (hunk 1 above)
Comment (destogl) [reply ros2_control_demos/package.xml]: Done

**Comment by destogl** [comment]
@olivier-stasse could you please review this PR?

**Comment by destogl** [comment]
I removed `upload-artifact` part for Coverage action. This is braking in all repositories and I am not sure why... For what are they used at all?

@bmagyar @olivier-stasse can you please review PR? Then we can finally merge :)

File: ros2_control_demo_communication_headless/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  set(CMAKE_C_STANDARD 99)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
```
Comment (olivier-stasse) [inline ros2_control_demo_communication_headless/CMakeLists.txt:15]: Same than previous comment : Why forcing the compiling option and not let the user specify it through colcon --cmake-args ?

File: ros2_control_demo_communication_headless/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+endif()
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+# uncomment the following section in order to fill in
+# further dependencies manually.
+# find_package(<dependency> REQUIRED)
```
Comment (olivier-stasse) [inline ros2_control_demo_communication_headless/CMakeLists.txt:22]: Remove three lines of comment

File: ros2_control_demo_hardware/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  set(CMAKE_C_STANDARD 99)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
```
Comment (olivier-stasse) [inline ros2_control_demo_hardware/CMakeLists.txt:15]: Same

File: ros2_control_demo_hardware/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+endif()
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+# uncomment the following section in order to fill in
+# further dependencies manually.
+# find_package(<dependency> REQUIRED)
```
Comment (olivier-stasse) [inline ros2_control_demo_hardware/CMakeLists.txt:22]: Same

File: ros2_control_demo_communication_gazebo/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  set(CMAKE_C_STANDARD 99)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
```
[the remark this answers is not in the document]
Comment (destogl) [reply ros2_control_demo_communication_gazebo/CMakeLists.txt:15]: This is a default from the package generation script. I have no idea what is the best way.

@bmagyar @Karsten1987 Do you have any opinion on that?

File: ros2_control_demo_communication_gazebo/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+endif()
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+# uncomment the following section in order to fill in
+# further dependencies manually.
+# find_package(<dependency> REQUIRED)
```
[the remark this answers is not in the document]
Comment (destogl) [reply ros2_control_demo_communication_gazebo/CMakeLists.txt:22]: Done

**Comment by olivier-stasse** [comment]
> > Looking forward to some code, now that we've got 1k lines of boilerplate stuff already ;)
> 
> I will not any code here just, configurations for the repo :)
> 
> I think we have every thing now, just want to make all tests happy....

IMHO, I think the point of @bmagyar is just to have codecov passing the test. If there is no test, codecov would failed (I did not test it in this configuration). It might be seen as testing the infra structure. I got your point about not having code, but an empty test would do it for me.

I do not have any strong opinion on this either option is fine with me.

File: ros2_control_demo_communication_gazebo/CMakeLists.txt (hunk 6 above)
[the remark this answers is not in the document]
Comment (bmagyar) [reply ros2_control_demo_communication_gazebo/CMakeLists.txt:15]: My personal opinion is that one should always aim to minimize the amount of cmake they write,and especially the cmake code they copy around. Unfortunately in ROS2 there's a lot of cargo cult stuff for Windows support. 

I'd actually be interested to see whether the linters pick up the same errors...

[the remark this answers is not in the document]
Comment (destogl) [reply ros2_control_demo_communication_gazebo/CMakeLists.txt:15]: > I'd actually be interested to see whether the linters pick up the same errors...

But they don't, at least in this package. Is there anything specifically to clarify?

**Comment by destogl** [comment]
> There's nothing blocking here but please open two follow up tickets:
>     * codecov

What do you mean exactly? I added dummy tests already.

**Pull request merged**